### PR TITLE
Leeroy/ug refactor

### PR DESCRIPTION
**Pull request by leeroy999**
- Refactor UserGuide.md
- Add PPP
- Refactor UI to not show selected event/member
- Refactor addressbook to ailurus

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -54,75 +56,87 @@ open the help window.<br>
...
   e.g. in `madd /n NAME`, `NAME` is a parameter which can be used as `madd /n John Doe`.
 
 * Items in square brackets are optional.<br>
   e.g. `/n NAME [/p POSITION]` can be used as `/n John Doe /p friend` or as `/n John Doe`.
 
 * Items with `…`​ after them can be used multiple times including zero times.<br>
-  e.g. `[/p POSITION]…​` can be used as ` ` (i.e. 0 times), `/p friend`, `/p friend /p family` etc.
+  e.g. `[/p POSITION]…​` can be completely absent (i.e. 0 times), `/p friend`, `/p friend /p family` etc.
+
+* All prefixes that are provided must be followed by a space and a non-empty parameter, except `/p ` prefix in `medit`, `/m ` prefix in `eedit`, `/att` and `/abs` in `mlist`, and `/ovd` in `tlist`.
```
Comment (GWphua) [inline docs/UserGuide.md:68]: Maybe can be written as 

* All prefixes that are provided must be followed by a space and a non-empty parameter.
There are exceptions to this rule :
1. XXXX
2. XXXX
3. XXXX

File: docs/UserGuide.md [hunk 2]
```diff
@@ -54,75 +56,87 @@ open the help window.<br>
...
-* Extraneous parameters for commands that do not take in parameters (such as `help`, `list` and `exit`) will be ignored.
+* Extraneous parameters for commands that do not take in parameters (such as `help`, `elist` and `exit`) will be ignored.
   <br>
   e.g. if the command specifies `help 123`, it will be interpreted as `help`.
 
-* Date and time will be rounded, such as: <br> <span id = 'id_date'>
-  1. if you specify date as `31/11/2021`, Ailurus will treat it as a valid date and round it to `30/11/2021`, while `32/11/2021` will be considered as an invalid date.
-  2. if you specify date as `29/02/2021`, Ailurus will round it to `28/02/2021` because 2021 is not a leap year.
-  3. Time `24:00` will be considered as `00:00` of the next valid day. If you specify a date time as `10/11/2021 24:00`, Ailurus will round it to `11/11/2021 00:00`.
+* If an invalid prefix is used in a command, it will be seen as part of the previous parameter and may throw an error.
```
Comment (Samuel-bit-prog) [inline docs/UserGuide.md:80]: add example

File: docs/UserGuide.md [hunk 3]
```diff
@@ -54,75 +56,87 @@ open the help window.<br>
 
 * Commands are usually abbreviated to facilitate faster typing for user convenience. As a general rule of thumb, `m`, `e` and `t` are used to represent `Member`, `Event` and `Task` respectively.
 
-* Words in `UPPER_CASE` are the parameters to be supplied by the user.<br>
+* Words in `UPPER_CASE` are the parameters to be supplied by the user, and they are preceded with prefixes of format `/prefix `<br>
```
Comment (Samuel-bit-prog) [inline docs/UserGuide.md:59]: change to previous

File: docs/UserGuide.md [hunk 4]
```diff
@@ -137,12 +151,11 @@ Edits an existing member in Ailurus.
...
 Format: `medit /m MEMBER_INDEX [/n NAME] [/ph PHONE] [/em EMAIL] [/a ADDRESS] [/p POSITION]…​`
 
-* `MEMBER_INDEX` **must be a positive integer** 1, 2, 3, …​
-* Edits the member at the specified `MEMBER_INDEX`. The index refers to the index number shown in the displayed member list.
+* Edits the member at the specified `MEMBER_INDEX`.
 * At least one of the optional fields must be provided.
 * Existing values will be updated to the input values.
 * When editing positions, the existing positions of the member will be removed i.e. adding of positions is not cumulative.
-* You can remove all the member’s positions by typing `/p` without specifying any positions after it.
+* You can remove all the member’s positions by typing `/p` without specifying any positions after it. In such a case, only one `/p` prefix is allowed, otherwise the second `/p` prefix will be considered an _invalid prefix_.
```
Comment (Samuel-bit-prog) [inline docs/UserGuide.md:158]: link back

File: docs/UserGuide.md [hunk 5]
```diff
@@ -54,75 +56,87 @@ open the help window.<br>
...
+#### Member Glossary
 
-### Member Commands
+Parameter | Description, Constraints
+--------|------------------
+**NAME** | _Name of member_ <br> - Only alphabets, numbers and spaces allowed i.e. no special characters such as `.`, `'`, or `-`. <br> - No character limit. <br> - All names of members must be unique <br> - If there are 2 members of the same name, it is recommended to add a number behind to uniquely identify the member e.g. John Tan2.
+**POSITIONS** | _Positions of member_ <br> - Same as NAME. <br> - If `-` should be used, replace it with a space e.g. `Vice President`.
+**PHONE_NUMBER** | _Phone number of member_ <br> - Only numbers allowed. <br> - No number limit, but minimum 3 numbers.
+**ADDRESS** | _Physical address of member_ <br> - All characters allowed with no character limit.
+**EMAIL** | _Email address of member_ <br> - Should be of format local-part@domain.
```
Comment (GWphua) [inline docs/UserGuide.md:110]: Maybe give an example of local-part@domain.

e.g. leeroy@example.com

File: docs/UserGuide.md (hunk 4 above)
Comment (GWphua) [inline docs/UserGuide.md:158]: Maybe give an example, needed to re-read for my first read.

**Review by Samuel-bit-prog — APPROVED** [review]
LGTM. just do some minor changes

**Review by GWphua — CHANGES_REQUESTED** [review]
Small changes needed, but LGTM.

**Review by GWphua — APPROVED** [review]
LGTM

**Comment by leeroy999** [comment]
Fixes #176, #180, #182, #185, #189, #192, #208, #221, #223, #229, #232, #247

**Pull request merged**